apple/OmnivoreKit/Sources/App/Views/Home/HomeFeedViewModel.swift

```diff
@@ -91,49 +91,53 @@ import Views
     items.insert(item, at: 0)
   }
 
-  func loadItems(dataService: DataService, audioController: AudioController, isRefresh: Bool) async {
-    let syncStartTime = Date()
-    let thisSearchIdx = searchIdx
-    searchIdx += 1
-
-    isLoading = true
-    showLoadingBar = true
-
+  func loadCurrentViewer(dataService: DataService) async {
     // Cache the viewer
     if dataService.currentViewer == nil {
-      Task { _ = try? await dataService.fetchViewer() }
+      _ = try? await dataService.fetchViewer()
     }
+  }
 
-    // Fetch labels if none are available locally
+  func loadLabels(dataService: DataService) async {
     let fetchRequest: NSFetchRequest<Models.LinkedItemLabel> = LinkedItemLabel.fetchRequest()
     fetchRequest.fetchLimit = 1
 
     if (try? dataService.viewContext.count(for: fetchRequest)) == 0 {
       _ = try? await dataService.labels()
     }
+  }
 
-    // Sync items if necessary
+  func syncItems(dataService: DataService, syncStartTime: Date) async {
     let lastSyncDate = dateFormatter.date(from: lastItemSyncTime) ?? Date(timeIntervalSinceReferenceDate: 0)
     let syncResult = try? await dataService.syncLinkedItems(since: lastSyncDate, cursor: nil)
     if syncResult != nil {
       lastItemSyncTime = dateFormatter.string(from: syncStartTime)
     }
+    // If possible start prefetching new pages in the background
+    if let itemIDs = syncResult?.updatedItemIDs,
+       let username = dataService.currentViewer?.username,
+       itemIDs.count > 0
+    {
+      Task {
```

Comment:
Looks like the Task block isn't needed now that the function is marked as async.

Reply:
What I was trying to achieve here is allowing the function to return without waiting on this call, but I might be misunderstanding what happens here.